src/main/scala/com/campudus/tableaux/TableauxRouter.scala

```diff
@@ -28,73 +26,45 @@ class TableauxRouter(verticle: Starter) extends Router with VertxAccess {
-        j <- controller.getJson(req).map(js => js.putString("action", "getColumn").putNumber("tableId", tableId.toLong).putNumber("columnId", columnId.toLong))
-        x <- controller.getColumn(j)
+        json <- getJson(req)
+        dbType <- Future.successful(Mapper.getDatabaseType(json.getString("type")))
+        x <- dbType match {
+          case "link" => controller.createColumn(tableId.toLong, json.getString("columnName"), dbType, json.getLong("toTable"), json.getLong("toColumn"), json.getLong("fromColumn"))
+          case _      => controller.createColumn(tableId.toLong, json.getString("columnName"), dbType)
+        }
       } yield x
     }
-    case Post("/tables") =>
-      AsyncReply {
-        for {
-          j <- controller.getJson(req).map(js => js.putString("action", "createTable"))
-          x <- controller.createTable(j)
-        } yield x
-      }
-    case Post(tableIdColumns(tableId)) =>
-      AsyncReply {
-        for {
-          j <- controller.getJson(req).map(js => js.putString("action", "createColumn").putNumber("tableId", tableId.toLong))
-          x <- controller.createColumn(j)
-        } yield x
-      }
-    case Post(tableIdRows(tableId)) =>
-      AsyncReply {
-        for {
-          j <- controller.getJson(req).map(js => js.putString("action", "createRow").putNumber("tableId", tableId.toLong))
-          x <- controller.createRow(j)
-        } yield x
-      }
-    case Post(tableIdColumnsIdRowsId(tableId, columnId, rowId)) =>
-      AsyncReply {
-        for {
-          j <- controller.getJson(req).map(js => js.putString("action", "fillRow").putNumber("tableId", tableId.toLong).putNumber("columnId", columnId.toLong).putNumber("rowId", rowId.toLong))
-          x <- controller.fillCell(j)
-        } yield x
-      }  
-    // more posts
-    case Get("/tables") => // list all tables
-      AsyncReply {
-        Future.successful(
-          Ok(
-            Json.obj(
-              "id" -> 123,
-              "name" -> "Personen",
-              "cols" -> Json.arr(
-                Json.obj("id" -> 1, "name" -> "Vorname"),
-                Json.obj("id" -> 2, "name" -> "Wohnort")),
-              "rows" -> Json.arr(
-                Json.obj("id" -> 1, "c1" -> "Peter", "c2" -> "Berlin"),
-                Json.obj("id" -> 2, "c1" -> "Kurt", "c2" -> "Düsseldorf")))))
+    case Post(tableIdRows(tableId)) => getAsyncReply(controller.createRow(tableId.toLong))
+    case Post(tableIdColumnsIdRowsId(tableId, columnId, rowId)) => getAsyncReply {
+      getJson(req) flatMap { json => controller.fillCell(tableId.toLong, columnId.toLong, rowId.toLong, json.getString("type"), json.getField("value")) }
+    }
+    case Delete(tableId(tableId))                    => getAsyncReply(controller.deleteTable(tableId.toLong))
+    case Delete(tableIdColumnsId(tableId, columnId)) => getAsyncReply(controller.deleteColumn(tableId.toLong, columnId.toLong))
+  }
+
+  private def getAsyncReply(f: Future[_]): AsyncReply = {

```

Comment:
Same as above -> Why not always `Future[DomainObject]`? Btw `DomainObject` is not a great name... :)